### ISSUE TITLE
Very unhelpful reporting of HTTP::Tiny internal errors

### ISSUE THREAD

**Issue by ap** [Leont/cpan-upload-tiny#2]
I just did this to get at the exception:

````diff
diff --git i/lib/CPAN/Upload/Tiny.pm w/lib/CPAN/Upload/Tiny.pm
index 504cba8c5b1..e29b122690b 100644
--- i/lib/CPAN/Upload/Tiny.pm
+++ w/lib/CPAN/Upload/Tiny.pm
@@ -46,7 +46,7 @@ sub upload_file {
                SUBMIT_pause99_add_uri_httpupload => ' Upload this file from my disk ',
        });

-       die "Upload failed: $result->{reason}\n" if !$result->{success};
+       die "Upload failed: $result->{ 599 eq $result->{status} ? 'content' : 'reason' }\n" if !$result->{success};

        return;
 }
````

I don’t know if you like the way I wrote that (and it’s just one line) so I didn’t make a full PR, but, something along those lines would be good.

**Comment by ap** [comment]
Or maybe like this, I dunno: 

````diff
diff --git i/lib/CPAN/Upload/Tiny.pm w/lib/CPAN/Upload/Tiny.pm
index 504cba8c5b1..0bf0450cb14 100644
--- i/lib/CPAN/Upload/Tiny.pm
+++ w/lib/CPAN/Upload/Tiny.pm
@@ -46,9 +46,10 @@ sub upload_file {
                SUBMIT_pause99_add_uri_httpupload => ' Upload this file from my disk ',
        });

-       die "Upload failed: $result->{reason}\n" if !$result->{success};
+       return if $result->{success};

-       return;
+       my $key = 599 eq $result->{status} ? 'content' : 'reason';
+       die "Upload failed: $result->{$key}\n";
 }

 sub read_config_file {
````

**Comment by ap** [comment]
(Fixed in 987d0814c5014d50222b44a0be13cd0eaa683c37. Thanks!)
